### DISCUSSION TITLE
Helidon WebClient typed collections serialization/deserialization examples

### DISCUSSION THREAD

**Discussion by RassulYunussov** [helidon-io/helidon#8530] [Q&A] [answered]
In essence I would like to obtain the same scenario as we have for a single element but with typed collections.

If we have something like this for a single element

```
WebClient webClient = WebClient.builder()
            .baseUri("http://localhost:8084")
            .build();

Product product = webClient.get()
                            .path("/product")
                            .requestEntity(Product.class);
```

How should it be for List<Product>?

Considering usage of helidon-http-media-jackson

**Comment by ljnelson** [comment, marked as answer]
(Disclaimer: I confess that the fluent style of APIs is never obvious to me; I'm an old-school javadoc-first person.)

Working backwards, you want a [`GenericType`](https://helidon.io/docs/latest/apidocs/io.helidon.common/io/helidon/common/GenericType.html), I think. Its documentation is very difficult to understand, but I think you create one like this (note the anonymous subclass nature of it):
```
new GenericType<List<Product>>() {};
```
So, untested, but does:
```
webClient.get()
    .path("/product")
    .request()
    .entity()
    .as(new GenericType<List<Product>>() {});
```
…get you pointed in the right direction? Or am I way off?

**Reply by RassulYunussov** [reply]
That works, thank you!
